packages/backend/src/core/EmailService.ts

```diff
@@ -208,19 +244,14 @@ export class EmailService {
 				blacklist: 'blacklist',
 			};
 
-			return {
-				available: false,
-				reason: validated.reason ? formatReason[validated.reason] ?? null : null,
-			};
-		}
-
-		const emailDomain: string = emailAddress.split('@')[1];
-		const isBanned = this.utilityService.isBlockedHost(this.meta.bannedEmailDomains, emailDomain);
+			// 自動追加が有効な場合はブラックリストに追加する
+			if (this.meta.enableAutoAddBannedEmailDomain) {
+				await this.addBlockedHost(emailDomain);
```

Comment:
残念ながら、上記のサービスの API キーを持っていないので、奇妙な入力 (非常に長いメールなど) に対してどのように反応するか、またこのドメインが「.com」のような形式になるかどうかはわかりません。安全なアプローチは、すべての自動禁止が eTLD の拡張であることを確認するために eTLD リストを使用することだと思いますが、個人的にはこのロジックに明らかな問題は見当たりません。

Unfortunately I don't have an API key to the services above so I am not sure how they would react to weird inputs. I think a safe approach is to use the eTLD list to make sure every auto ban is an extension of an eTLD, but it might not be actually a problem depending on the actual response from these third-party APIs:

https://developer.mozilla.org/en-US/docs/Glossary/eTLD